new scipy release (1.7.0) breaks KNN tests

SciPy 1.7.0 was [released](https://github.com/scipy/scipy/releases/tag/v1.7.0) yesterday with changes to `cdist`.  It seems we need to remove an arg from our KNN converters, see [build](https://github.com/microsoft/hummingbird/runs/2877767604?check_suite_focus=true).

Ex:      `TypeError: cdist_old_weighted_minkowski_double_wrap() takes at most 5 arguments (6 given)`

for:
```
=========================== short test summary info ============================
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_mahalanobis
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_seuclidean
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_classifer_wminkowski
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_mahalanobis
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_seuclidean
FAILED tests/test_sklearn_kneighbors.py::TestSklearnKNeighbors::test_kneighbors_regressor_wminkowski
===== 6 failed, 333 passed, 263 skipped, 95 warnings in 101.69s (0:01:41) ======
```

Troubleshooting in #526...I pinned just to verify that it's the new scipy version that breaks it, then we need to fix the args, unpin, and push.

We need to fix this before we merge #525.

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -54,9 +54,6 @@ jobs:
       run: |
         pip install .[tests] -f https://download.pytorch.org/whl/torch_stable.html
         pip install git+https://github.com/onnx/sklearn-onnx.git@9b551d8ecfd9f97ad8771909e34cfffea7b6e99d
-    - name: Downgrade scipy for now (see issue 525)
-      if:  matrix.python-version > '3.6'
-      run: pip install scipy==1.6.3
     - name: Run basic tests without extra
       run: pytest
     - name: Coverage on basic tests without extra

--- a/tests/test_sklearn_kneighbors.py
+++ b/tests/test_sklearn_kneighbors.py
@@ -27,13 +27,12 @@ class TestSklearnKNeighbors(unittest.TestCase):
             X = X.astype(np.float32)
 
             if metric == "wminkowski":
-                metric_params["w"] = np.random.rand(X.shape[1])
+                metric_params = {"w": np.random.rand(X.shape[1])}
             elif metric == "seuclidean":
-                metric_params["V"] = np.random.rand(X.shape[1])
+                metric_params = {"V": np.random.rand(X.shape[1])}
             elif metric == "mahalanobis":
                 V = np.cov(X.T)
-                metric_params["V"] = V
-                metric_params["VI"] = np.linalg.inv(V)
+                metric_params = {"VI": np.linalg.inv(V)}
 
             model = KNeighborsClassifier(
                 n_neighbors=n_neighbors, algorithm=algorithm, weights=weights, metric=metric, metric_params=metric_params
@@ -118,12 +117,12 @@ class TestSklearnKNeighbors(unittest.TestCase):
             X = X.astype(np.float32)
 
             if metric == "wminkowski":
-                metric_params["w"] = np.random.rand(X.shape[1])
+                metric_params = {"w": np.random.rand(X.shape[1])}
             elif metric == "seuclidean":
-                metric_params["V"] = np.random.rand(X.shape[1])
+                metric_params = {"V": np.random.rand(X.shape[1])}
             elif metric == "mahalanobis":
                 V = np.cov(X.T)
-                metric_params["VI"] = np.linalg.inv(V)
+                metric_params = {"VI": np.linalg.inv(V)}
 
             model = KNeighborsRegressor(
                 n_neighbors=n_neighbors, algorithm=algorithm, weights=weights, metric=metric, metric_params=metric_params

--- a/tests/test_sklearn_kneighbors.py
+++ b/tests/test_sklearn_kneighbors.py
@@ -112,8 +112,9 @@ class TestSklearnKNeighbors(unittest.TestCase):
         metric_params={"p": 2},
         score_w_train_data=False,
     ):
-        for data in [datasets.load_boston(), datasets.load_diabetes()]:
-            X, y = data.data, data.target
+        for data in [datasets.fetch_california_housing(), datasets.load_diabetes()]:
+            # take all of diabetes and a subset of housing (which is slow to test otherwise)
+            X, y = data.data[0:441], data.target
             X = X.astype(np.float32)
 
             if metric == "wminkowski":

